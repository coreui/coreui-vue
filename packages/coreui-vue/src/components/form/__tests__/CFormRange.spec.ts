import { mount } from '@vue/test-utils'
import { CFormRange as Component } from '../../../index'

const ComponentName = 'CFormRange'

const defaultWrapper = mount(Component, {
  propsData: {},
  slots: {
    default: 'Default slot',
  },
})

const customWrapper = mount(Component, {
  propsData: {
    disabled: true,
    max: 400,
    min: 50,
    readonly: true,
    steps: 10,
    value: 250,
  },
  slots: {
    default: 'Default slot',
  },
})

describe(`Loads and display ${ComponentName} component`, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.html()).toMatchSnapshot()
  })
  it('contain slots and classes', () => {
    expect(defaultWrapper.text()).toContain('Default slot')
    expect(defaultWrapper.find('input').classes('form-range')).toBe(true)
    expect(defaultWrapper.find('input').attributes('type')).toBe('range')
  })
})

describe(`Customize ${ComponentName} component`, () => {
  it('renders correctly', () => {
    expect(customWrapper.html()).toMatchSnapshot()
  })
  it('contain slots and classes', () => {
    expect(customWrapper.text()).toContain('Default slot')
    expect(customWrapper.find('input').classes('form-range')).toBe(true)
    expect(customWrapper.find('input').attributes('type')).toBe('range')
    expect(customWrapper.find('input').attributes('disabled')).toBe('')
    expect(customWrapper.find('input').attributes('max')).toBe('400')
    expect(customWrapper.find('input').attributes('min')).toBe('50')
    expect(customWrapper.find('input').attributes('readonly')).toBe('')
    expect(customWrapper.find('input').attributes('steps')).toBe('10')
  })
})
