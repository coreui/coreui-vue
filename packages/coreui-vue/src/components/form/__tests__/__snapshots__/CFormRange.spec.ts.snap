// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customize CFormRange component renders correctly 1`] = `"<input class=\\"form-range\\" disabled=\\"\\" max=\\"400\\" min=\\"50\\" readonly=\\"\\" steps=\\"10\\" type=\\"range\\">"`;

exports[`Loads and display CFormRange component renders correctly 1`] = `"<input class=\\"form-range\\" type=\\"range\\">"`;
